### PR TITLE
vault: update authentication func

### PR DESCRIPTION
**Pull request by saheienko**
Signed-off-by: Serhii Aheienko <serhii.aheienko@gmail.com>

**What this PR does / why we need it**
For now if both `VAULT_TOKEN` and `VAULT_AUTH_METHOD` parameters are provided, the last is used. Change it to the `VAULT_TOKEN` with this change.

**Which issue(s) this PR fixes** (optional)
Closes #

**Special notes for your reviewer**:

### CODE REVIEW

File: vault/utils/utils.go [hunk 1]
```diff
@@ -36,8 +36,8 @@ const (
 )
 
 var (
-	ErrVaultTokenNotSet    = errors.New("VAULT_TOKEN not set.")
-	ErrVaultAddressNotSet  = errors.New("VAULT_ADDR not set.")
+	ErrVaultTokenNotSet    = errors.New("VAULT_TOKEN or VAULT_AUTH_METHOD not set")
```
Comment (leseb) [inline vault/utils/utils.go:39]: Shouldn't the name of the variable be changed as well?

Comment (saheienko) [reply vault/utils/utils.go]: Thanks. Changed to `ErrVaultAuthParamsSet`.

**Review by leseb — COMMENTED** [review]
Just a general question, what's the rationale for picking token first over Kubernetes auth? Using Kubernetes auth seems easier from a management standpoint (all JWT are rotated/renewed internally without intervention), for token they would either need to have their TTL extended or created in a way that they never expire. Thanks.

**Review by adityadani — APPROVED** [review]

**Comment by adityadani** [comment]
@leseb This client tries to mimic vault client's (CLI) behavior, which gives VAULT_TOKEN the highest preference over any other vault auth methods.

**Pull request merged**